### PR TITLE
Changed to throw a more standard error instead of a PlaidError for ne…

### PR DESCRIPTION
**Pull request by edmundloo**
…twork failure

### CODE REVIEW

File: lib/plaid/connect.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def self.run(http, request)
       response = http.request(request)
 
       if response.body.nil? || response.body.empty?
-        raise Plaid::PlaidError.new(0, 'Server error', 'Try to connect later')
+        raise 'Server error, try to connect again later.'
```
Comment (be9) [inline lib/plaid/connect.rb]: Raising `String` objects from libraries is a bad practice. You ought to add another error subclass in `lib/plaid/errors.rb` and raise it.

Ideally, we should have a inheritance like this:

```ruby
class PlaidError < StandardError; end

class PlaidServerError < PlaidError; end

class PlaidAPIError < PlaidError
  attr_reader :error_type, :error_code, :error_message, :display_message, :request_id

  # ... other bits of what is now called PlaidError
end
```

Given this, a client could handle all possible errors by:

```ruby
begin
   # Call Plaid::Whatever.methods
rescue Plaid::PlaidError => e
  # Catch every possible error raised by plaid gem
end
```

**Comment by edmundloo** [comment]
@be9 : Updated! :)

File: lib/plaid/connect.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@ def self.run(http, request)
       response = http.request(request)
 
       if response.body.nil? || response.body.empty?
-        raise Plaid::PlaidError.new(0, 'Server error', 'Try to connect later')
+        raise(PlaidServerError, 'Server error, try to connect again later.')
```
Comment (be9) [inline lib/plaid/connect.rb]: Parens are not needed here. Check the [style guide](https://github.com/bbatsov/ruby-style-guide#exceptions).

Comment (edmundloo) [inline lib/plaid/connect.rb]: 🐳 Thanks Oleg!

**Pull request merged**